
Instrumented the sparc assembly to compile for sparc64 with recent binutils
versions ...

  - Rene Rebe <rene@exactcode.de>

diff -ur libgcrypt-1.2.1/mpi/sparc32/mpih-add1.S libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-add1.S
--- libgcrypt-1.2.1/mpi/sparc32/mpih-add1.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-add1.S	2005-01-23 06:34:02.000000000 +0100
@@ -39,6 +39,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(_gcry_mpih_add_n)
diff -ur libgcrypt-1.2.1/mpi/sparc32/mpih-lshift.S libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-lshift.S
--- libgcrypt-1.2.1/mpi/sparc32/mpih-lshift.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-lshift.S	2005-01-23 06:34:24.000000000 +0100
@@ -29,6 +29,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(_gcry_mpih_lshift)
diff -ur libgcrypt-1.2.1/mpi/sparc32/mpih-rshift.S libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-rshift.S
--- libgcrypt-1.2.1/mpi/sparc32/mpih-rshift.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32/mpih-rshift.S	2005-01-23 06:36:43.000000000 +0100
@@ -28,6 +28,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align	4
 	.global C_SYMBOL_NAME(_gcry_mpih_rshift)
diff -ur libgcrypt-1.2.1/mpi/sparc32/udiv.S libgcrypt-1.2.1-fixed/mpi/sparc32/udiv.S
--- libgcrypt-1.2.1/mpi/sparc32/udiv.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32/udiv.S	2005-01-23 06:36:59.000000000 +0100
@@ -36,6 +36,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 	.text
 	.align 4
 	.global C_SYMBOL_NAME(__udiv_qrnnd)
diff -ur libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul1.S libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul1.S
--- libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul1.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul1.S	2005-01-23 06:32:51.000000000 +0100
@@ -37,6 +37,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align	8
 	.global C_SYMBOL_NAME(_gcry_mpih_mul_1)
diff -ur libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul2.S libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul2.S
--- libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul2.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul2.S	2005-01-23 06:33:07.000000000 +0100
@@ -37,6 +37,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align 4
 	.global C_SYMBOL_NAME(_gcry_mpih_addmul_1)
diff -ur libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul3.S libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul3.S
--- libgcrypt-1.2.1/mpi/sparc32v8/mpih-mul3.S	2002-05-14 15:11:08.000000000 +0200
+++ libgcrypt-1.2.1-fixed/mpi/sparc32v8/mpih-mul3.S	2005-01-23 06:33:51.000000000 +0100
@@ -36,6 +36,9 @@
 
 #include "sysdep.h"
 
+	.register %g2,#scratch
+	.register %g3,#scratch
+
 .text
 	.align 4
 	.global C_SYMBOL_NAME(_gcry_mpih_submul_1)
